packages/cli-repl/src/arg-mapper.ts

```diff
@@ -103,6 +104,25 @@ export function applyTlsCertificateSelector(
   }
 }
 
+/**
+   * Applies driverInfo to track usage of mongosh,
+   * because appName can be overwritten by a user in their connection string.
+   *
+   * @param {MongoClientOptions} nodeOptions - The MongoClient options.
+   *
+   * @returns {DriverInfo} The driverInfo metadata.
+   */
+export function applyDriverInfo(
+  nodeOptions: MongoClientOptions
+): void {
+  const { version } = require('../package.json');
+  nodeOptions.driverInfo = {
+    name: 'mongosh',
+    version,
+    platform: process.platform
```

Comment:
This is not quite the idea behind `platform` – it doesn’t provide the OS, it provides the runtime that the user is connecting from.

This is what a connection attempt currently looks like in the server logs:

```js
$ cat mongod.log | tail -n1 | json_pp
{
   "attr" : {
      "client" : "conn40",
      "doc" : {
         "application" : {
            "name" : "mongosh 1.1.2" // <-- from appName
         },
         "driver" : {
            "name" : "nodejs", // <-- will become "nodejs|mongosh" with this patch
            "version" : "4.1.4" // <-- will become e.g. "4.3.0|1.1.9" with this patch
         },
         "os" : {
            "architecture" : "x64",
            "name" : "linux",
            "type" : "Linux",
            "version" : "5.11.0-46-generic"
         },
         "platform" : "Node.js v14.18.0, LE (unified)|Node.js v14.18.0, LE (unified)"
      },
      "remote" : "127.0.0.1:35364"
   },
   "c" : "NETWORK",
   "ctx" : "conn40",
   "id" : 51800,
   "msg" : "client metadata",
   "s" : "I",
   "t" : {
      "$date" : "2022-01-13T10:47:01.451+01:00"
   }
}
```

Reply:
I think I can omit the platform then since it is not a required property.